src/DurableTask.SqlServer/SqlUtils.cs

```diff
@@ -134,7 +134,7 @@ public static HistoryEvent GetHistoryEvent(this DbDataReader reader, bool isOrch
                     historyEvent = new OrchestratorStartedEvent(eventId);
                     break;
                 case EventType.SubOrchestrationInstanceCompleted:
-                    historyEvent = new SubOrchestrationInstanceCompletedEvent(eventId, GetTaskId(reader), GetPayloadText(reader));
+                    historyEvent = new SubOrchestrationInstanceCompletedEvent(eventId: -1, GetTaskId(reader), GetPayloadText(reader));
```

Comment:
Specifying -1 for the event ID is how these events are normally created in DurableTask.Core. The fact that were were initializing these events with different values was confusing certain code within DurableTask.Core which was overly sensitive to their specific values.

Reply:
Can you share a link to the code in DurableTask.Core where the event ID confusion happened?

Reply:
Sure - it's the code in [the other PR: TaskOrchestrationDispatcher.ReconcileMessagesWithState](https://github.com/Azure/durabletask/pull/1018/files).

Basically, we were iterating through the orchestration history looking for an event with a particular ID and then casting it to a particular type. Normally this works fine because each event in the history has a unique event ID number (except for the ones with -1). The DT.Core code assumed that EventId values would be unique. However, the bug fixed by this PR was that we were setting multiple (related) events with the same Event ID value, which is why we were casting history events to the wrong type as shown in the error message for that PR.